### PR TITLE
Lf 3215 a user creating a task at a location concurrently with another user retiring said location creates a task at a location that does not exist

### PR DESCRIPTION
**Pull request by gursimran-singh**
**Description**

When two users are logged on to the same farm and one user begins the task creation flow at a particular location while another user attempts to retire the same location, the tasks are created at a location that does not exist on the farm

Jira link:https://lite-farm.atlassian.net/browse/LF-3215

**Type of change**

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

- Start to create a new task
- Before finishing the task creation flow, delete the location by setting the "deleted = true" in database to mimic the behaviour that another user has deleted the location at the same time.
- Try to save the task. Error will occur and a modal will be displayed giving info of the error.

- [x] Passes test case
- [x] UI components visually reviewed on desktop view
- [x] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [x] I have added or updated language tags for text that's part of the UI
- [x] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

File: packages/webapp/public/locales/en/translation.json [hunk 1]
```diff
@@ -1894,7 +1894,11 @@
     "TODO": "To do",
     "TRANSPLANT": "Transplant",
     "TRANSPLANT_LOCATIONS": "Select a transplant location",
-    "UNASSIGNED": "Unassigned"
+    "UNASSIGNED": "Unassigned",
+    "CREATE": {
+      "FAILED": "Can't create the task",
+      "LOCATION_DELETED": "It seems you are trying to create a task on a deleted location. Please try again afresh!"
```
Comment (antsgar) [inline packages/webapp/public/locales/en/translation.json:1900]: I'd suggest

```suggestion
      "LOCATION_DELETED": "Another farmer has retired your selected location. Please refresh the page to get the most recent status."
```
so it's more consistent with a pre-existing string we have for attempting to delete a plan with completed tasks.

**Review by antsgar — APPROVED** [review]
It's working great, thank you so much for contributing! 🚀

File: packages/api/src/controllers/taskController.js [hunk 2]
```diff
@@ -270,6 +273,20 @@ const taskController = {
     const nonModifiable = getNonModifiable(typeOfTask);
     return async (req, res, next) => {
       try {
+        // Only to satisfy "isDeleted" function requirement
```
Comment (kathyavini) [inline packages/api/src/controllers/taskController.js:276]: If it's truly only for that and there is no other reason to open a transaction, I think you might be able to just pass `null` as the first argument to `baseController.isDeleted()` -- just tried that on another ticket and it seemed fine!

Comment (gursimran-singh) [reply packages/api/src/controllers/taskController.js]: @kathyavini Thanks for the suggestion.

**Review by antsgar — APPROVED** [review]

**Pull request merged**